com.unity.multiplayer.mlapi/Editor/CodeGen/XXHash/XXHash.cs

```diff
@@ -0,0 +1,291 @@
+// <copyright file="XXHash.cs" company="Sedat Kapanoglu">
+// Copyright (c) 2015-2019 Sedat Kapanoglu
+// MIT License (see LICENSE file for details)
+// </copyright>
+
+// @mfatihmar (Unity): Modified for Unity support
+
+using System.Runtime.CompilerServices;
```

Comment:
To @TwoTenPvP's point, I'd like to do one of these 4 things: 
1) re-use the hashing we already have in MLAPI
2) use XXHash to replace the hashing we already have in MLAPI
3) use a hashing scheme in standard c# / Unity
4) if none of the above is acceptable, call out in code somewhere why this specific hash algorithm had to be imported (e.g. what are its characteristics and how it relates to the application)

Reply:
e.g. "We needed a non-cryptographic speed-optimized hash", etc.

Reply:
I defer to PR #408 on this aspect.  I required that branch in order to complete the RPC Queue stuff.
However, if I were to guess why he was using a different hashing algorithm it would be relative to the following links:
[xxHash Benchmarks](https://cyan4973.github.io/xxHash/)
Further Details: [https://github.com/Cyan4973/xxHash](https://github.com/Cyan4973/xxHash)

So, I think there may be a case where it depends on what kind of has you are trying to generate.  Do you want a cryptographic safe (or safer) hash or do you want a small data super fast hash?

There very likely could be spaces for both FNV and XXHash in MLAPI where it depends upon the usage (secure vs fast) case that dictates which one to use?